### ISSUE TITLE
Web_1 Fails due to permission error Debian Jessie

### ISSUE THREAD

**Issue by relshire** [biosistemika/scinote-web#9]
Hi,

I have tried this on two debian jessie installs. Same result. Does not work even if paths are created manually, or permissions set to all r/w access. I followed the instructions in the main .md file and ran everything under root.

Here is the error message.

```
web_1     | => Booting Puma
web_1     | => Rails 4.2.5 application starting in development on http://0.0.0.0:3000
web_1     | => Run `rails server -h` for more startup options
web_1     | => Ctrl-C to shutdown server
web_1     | Exiting
web_1     | /usr/local/lib/ruby/2.2.0/fileutils.rb:252:in `mkdir': Permission denied @ dir_s_mkdir - /usr/src/app/tmp (Errno::EACCES)
web_1     |     from /usr/local/lib/ruby/2.2.0/fileutils.rb:252:in `fu_mkdir'
web_1     |     from /usr/local/lib/ruby/2.2.0/fileutils.rb:226:in `block (2 levels) in mkdir_p'
web_1     |     from /usr/local/lib/ruby/2.2.0/fileutils.rb:224:in `reverse_each'
web_1     |     from /usr/local/lib/ruby/2.2.0/fileutils.rb:224:in `block in mkdir_p'
web_1     |     from /usr/local/lib/ruby/2.2.0/fileutils.rb:210:in `each'
web_1     |     from /usr/local/lib/ruby/2.2.0/fileutils.rb:210:in `mkdir_p'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/server.rb:134:in `block in create_tmp_directories'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/server.rb:133:in `each'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/server.rb:133:in `create_tmp_directories'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/server.rb:77:in `start'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/commands_tasks.rb:80:in `block in server'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/commands_tasks.rb:75:in `tap'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/commands_tasks.rb:75:in `server'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
web_1     |     from /usr/local/bundle/gems/railties-4.2.5/lib/rails/commands.rb:17:in `<top (required)>'
web_1     |     from bin/rails:4:in `require'
web_1     |     from bin/rails:4:in `<main>'
scinoteweb_web_1 exited with code 1

```
Any assistance will be greatly appreciated.

--Rob

**Comment by jbargu** [comment]
Hi,

I believe this must be an issue running everything as root. Could you provide a list of exact commands you used to setup scinote (you can access them by using `history` command). Do you run docker as root as well?

**Comment by relshire** [comment]
Hi,

Docker is run as a service. 
`start service docker`
I set up docker for Debian Jessie using the instructions found here: https://docs.docker.com/engine/installation/linux/debian/
I tested the installation by running (as root) 
`docker run hello-world`
There were no errors.
I installed docker compose using the instructions found here: https://docs.docker.com/compose/install/

I created the application.yml file as described here: https://github.com/biosistemika/scinote-web

Then I ran the following commands as root:
```
make docker
make cli
make run
```
This produced the error message above. I have also tried changing permissions on the /usr/src/tmp to allow global r/w/x access. The error that it returns is the same as above.

Thanks again for you assistance.

--Rob

**Comment by jbargu** [comment]
Could you try adding normal user to `docker group` as described [here](https://docs.docker.com/engine/installation/linux/debian/#giving-non-root-access)? Then as a normal user run `git clone` in completely new folder and perform all the steps above from scratch (`make docker˙, ...) as a normal user as well. I believe root account is giving you headache. I personally run everything as normal user.

By the way, which folder did you chmod? /usr/src/app/tmp in docker is mapped to the tmp folder in scinote folder on host (not the actual /usr/src/app/tmp folder).

I hope this helps but please let me know if it doesn't work.

**Comment by relshire** [comment]
Hi,

I added a normal user that I created for this test (scinote) to the docker group. I then started from scratch and did all the steps as previously described. It failed in the same was as before.

I did the same thing with user number 1000 (the first user added to the system) and it worked fine. It seems that the actual system user number must match the scinote user number in the container.

Thanks for pointing me in the right direction.

--Rob
